lib/garage/config.rb

```diff
@@ -49,11 +49,13 @@ def docs
     end
 
     def cast_resource
-      @cast_resource ||= proc { |resource|
+      @cast_resource ||= proc { |resource, options|
+        options ||= {}
+        to_resource_call = [:to_resource, options[:cast_context]].compact
         if resource.respond_to?(:map) && resource.respond_to?(:to_a)
-          resource.map(&:to_resource)
+          resource.map { |r| r.__send__(*to_resource_call) }
```

Comment:
We can avoid `send` like below

```patch
diff --git a/lib/garage/config.rb b/lib/garage/config.rb
index 2402679..3ee64da 100644
--- a/lib/garage/config.rb
+++ b/lib/garage/config.rb
@@ -51,11 +51,11 @@ module Garage
     def cast_resource
       @cast_resource ||= proc { |resource, options|
         options ||= {}
-        to_resource_call = [:to_resource, options[:cast_context]].compact
+        to_resource_args = [options[:cast_context]].compact
         if resource.respond_to?(:map) && resource.respond_to?(:to_a)
-          resource.map { |r| r.__send__(*to_resource_call) }
+          resource.map { |r| r.to_resource(*to_resource_args) }
         else
-          resource.__send__(*to_resource_call)
+          resource.to_resource(*to_resource_args)
         end
       }
     end
```

Reply:
👍  I didn't know that splatting an empty array would be equivalent to passing no arguments (rather than nil), but that makes sense when I consider how splat is supposed to work.


```
irb(main):001:0> def to_resource
irb(main):002:1> end
=> :to_resource
irb(main):003:0> to_resource(1)
Traceback (most recent call last):
        5: from /Users/leonard-chin/.rbenv/versions/2.6/bin/irb:23:in `<main>'
        4: from /Users/leonard-chin/.rbenv/versions/2.6/bin/irb:23:in `load'
        3: from /Users/leonard-chin/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):3
        1: from (irb):1:in `to_resource'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):004:0> to_resource(*[])
=> nil
```